### PR TITLE
Linked API to Create School button, added tests and Storybook v.2

### LINKED ISSUE #7: Make the "create school" button work.

### LINKED ISSUE #8: Link the create school button to correct API and make a SchoolCreate.js page for that.

### PR DESCRIPTION
**Pull request by sherpatran**
Made sure that the CreateSchool button is working. Connected it to the proper Schools POST api and added the tests/storybook for that feature.

The parameter termRegex had to be encoded because of the disallowed characters "[" and "\". This allows termRegex to be passed through the frontend without any issues. Then, to get the accurate value passed into the backend, termRegex is decoded in the SchoolController.

https://organic-sherpatran-dev.dokku-11.cs.ucsb.edu/

Closes https://github.com/ucsb-cs156-s24/proj-organic-s24-5pm-3/issues/7 and closes https://github.com/ucsb-cs156-s24/proj-organic-s24-5pm-3/issues/8.

### CODE REVIEW

**Review by Xalex217 — COMMENTED** [review]
LGTM

**Review by Xalex217 — DISMISSED** [review]
lgtm

**Review by yuxiaolejs — DISMISSED** [review]
LGTM

**Review by pconrad — DISMISSED** [review]
Code LGTM.. will test on dokku

File: frontend/src/main/pages/SchoolCreatePage.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+export default function SchoolCreatePage({ storybook = false }) {
+  const objectToAxiosParams = (school) => ({
+    url: "/api/schools/post",
+    method: "POST",
+    params: {
+      abbrev: school.abbrev,
+      name: school.name,
+      termRegex: encodeURIComponent(school.termRegex),
+      termDescription: school.termDescription,
+      termError: school.termError,
```
Comment (pconrad) [inline frontend/src/main/pages/SchoolCreatePage.js:16]: I'm wondering if this parameter is still in the backend? 
And if it's required? And if the fact that it's been removed from the form is the reason for the error?

**Review by pconrad — CHANGES_REQUESTED** [review]
This is what happens when I try to test this PR

File: frontend/src/main/pages/SchoolCreatePage.js (hunk 1 above)
Comment (yuxiaolejs) [reply frontend/src/main/pages/SchoolCreatePage.js:16]: I asked Aaron and he said you asked him to remove it, but only for the frontend, and already merged into main in some previous PRs (which means main is probably broken now).

Comment (yuxiaolejs) [reply frontend/src/main/pages/SchoolCreatePage.js:16]: I have proposed a fix in #74

**Comment by yuxiaolejs** [comment]
I have merged 74 into this PR, so if this one gets merged, we just

Closes #74

**Pull request merged**